rules/windows/discovery_post_exploitation_external_ip_lookup.toml

```
@@ -0,0 +1,88 @@
+[metadata]
+creation_date = "2020/09/04"
+maturity = "production"
+updated_date = "2021/04/08"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies domains commonly used by adversaries for post-exploitation IP lookups. It is common for adversaries to
+test for Internet access and acquire their external IP address after they have gained access to a system. Among others,
+this has been observed in campaigns leveraging the information stealer, Trickbot.
+"""
+false_positives = [
+    """
+    If the domains listed in this rule are used as part of an authorized workflow, this rule will be triggered by those
+    events. Validate that this is expected activity and tune the rule to fit your environment variables.
+    """,
+]
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "External IP Lookup fron Non-Browser Process"
+references = [
+    "https://community.jisc.ac.uk/blogs/csirt/article/trickbot-analysis-and-mitigation",
+    "https://www.cybereason.com/blog/dropping-anchor-from-a-trickbot-infection-to-the-discovery-of-the-anchor-malware",
+]
+risk_score = 21
+rule_id = "1d72d014-e2ab-4707-b056-9b96abe7b511"
+severity = "low"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Discovery"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+network where network.protocol == "dns" and
+    process.name != null and user.id not in ("S-1-5-19", "S-1-5-20") and
+    event.action == "lookup_requested" and
+    /* Add new external IP lookup services here */
+    dns.question.name :
+    (
+        "*api.ipify.org",
+        "*freegeoip.app",
+        "*checkip.amazonaws.com",
+        "*checkip.dyndns.org",
+        "*freegeoip.app",
+        "*icanhazip.com",
+        "*ifconfig.*",
+        "*ipecho.net",
+        "*ipgeoapi.com",
+        "*ipinfo.io",
+        "*ip.anysrc.net",
+        "*myexternalip.com",
+        "*myipaddress.com",
+        "*showipaddress.com",
+        "*whatismyipaddress.com",
+        "*wtfismyip.com"
+    ) and
+    /* Insert noisy false positives here */
+    not process.executable :
+    (
+      "C:\\Program Files\\*.exe",
+      "C:\\Program Files (x86)\\*.exe",
+      "C:\\Windows\\System32\\WWAHost.exe",
+      "C:\\Windows\\System32\\smartscreen.exe",
+      "C:\\Windows\\System32\\MicrosoftEdgeCP.exe",
+      "C:\\ProgramData\\Microsoft\\Windows Defender\\Platform\\*\\MsMpEng.exe",
+      "C:\\Users\\*\\AppData\\Local\\Google\\Chrome\\Application\\chrome.exe",
+      "C:\\Users\\*\\AppData\\Local\\Programs\\Fiddler\\Fiddler.exe",
+      "C:\\Users\\*\\AppData\\Local\\Programs\\Microsoft VS Code\\Code.exe",
+      "C:\\Users\\*\\AppData\\Local\\Microsoft\\OneDrive\\OneDrive.exe"
```

```suggestion
      "?:\\Program Files\\*.exe",
      "?:\\Program Files (x86)\\*.exe",
      "?:\\Windows\\System32\\*.exe",
      "?:\\Windows\\SysWOW64\\*.exe",
      "?:\\ProgramData\\Microsoft\\Windows Defender\\Platform\\*\\MsMpEng.exe",
      "?:\\Users\\*\\AppData\\Local\\Google\\Chrome\\Application\\chrome.exe",
      "?:\\Users\\*\\AppData\\Local\\Programs\\Fiddler\\Fiddler.exe",
      "?:\\Users\\*\\AppData\\Local\\Programs\\Microsoft VS Code\\Code.exe",
      "?:\\Users\\*\\AppData\\Local\\Microsoft\\OneDrive\\OneDrive.exe"
```
I'm assuming it would be odd for exe's in sys32 and syswow64 to be reaching out?